### PR TITLE
Upgrade dependencies

### PR DESCRIPTION
**Pull request by VachaShah**
Signed-off-by: Vacha Shah <vachshah@amazon.com>

### Description
Upgrading dependencies for:

* hadoop3
* aws
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [X] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [X] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: buildSrc/build.gradle [hunk 1]
```diff
@@ -112,6 +112,7 @@ dependencies {
   api 'commons-io:commons-io:2.7'
   api "net.java.dev.jna:jna:5.11.0"
   api 'gradle.plugin.com.github.johnrengelman:shadow:7.1.2'
+  api 'org.jdom:jdom2:2.0.6.1'
```
Comment (reta) [inline buildSrc/build.gradle:115]: Sorry, was (is) this one transitive before?

File: server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java [hunk 2]
```diff
@@ -248,7 +248,7 @@ private void assertSegmentStats(int numberOfReplicas) throws IOException {
 
     /**
      * Waits until the replica is caught up to the latest primary segments gen.
-     * @throws Exception
+     * @throws Exception if assertion fails
```
Comment (reta) [inline server/src/internalClusterTest/java/org/opensearch/indices/replication/SegmentReplicationIT.java:251]: :+1: ty

File: buildSrc/build.gradle (hunk 1 above)
Comment (VachaShah) [reply buildSrc/build.gradle:115]: Yes, jdom2 comes from `gradle.plugin.com.github.johnrengelman:shadow:7.1.2`, so making sure it picks the latest jdom2 version.

**Comment by VachaShah** [comment]
Gradle check failure:

```
FAILURE: Build failed with an exception.

* What went wrong:
Could not receive a message from the daemon.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org/
Gradle Check Failed!
```

**Review by reta — APPROVED** [review]

File: plugins/repository-hdfs/licenses/jetty-server-LICENSE.txt [hunk 3]
```diff
@@ -0,0 +1,202 @@
+
```
Comment (dreamer-89) [inline plugins/repository-hdfs/licenses/jetty-server-LICENSE.txt:1]: @VachaShah @reta : Are there any guidelines for reviewing license content.

Comment (VachaShah) [reply plugins/repository-hdfs/licenses/jetty-server-LICENSE.txt:1]: I got the license from https://mvnrepository.com/artifact/org.eclipse.jetty/jetty-server/9.4.48.v20220622 Licenses section.

**Comment by VachaShah** [comment]
@reta Can you review this again please?

File: test/fixtures/hdfs-fixture/build.gradle [hunk 4]
```diff
@@ -44,4 +44,6 @@ dependencies {
   api "com.fasterxml.jackson.core:jackson-databind:${versions.jackson_databind}"
   api 'net.minidev:json-smart:2.4.8'
   api "org.mockito:mockito-core:${versions.mockito}"
+  api "com.google.protobuf:protobuf-java:3.21.2"
+  api "org.eclipse.jetty.websocket:websocket-client:9.4.48.v20220622"
```
Comment (reta) [inline test/fixtures/hdfs-fixture/build.gradle:48]: I believe this dependency could be excluded, there is no WS usage anywhere (afaik), right?

File: plugins/repository-hdfs/build.gradle [hunk 5]
```diff
@@ -83,6 +83,7 @@ dependencies {
   api 'net.minidev:json-smart:2.4.8'
   api 'org.apache.zookeeper:zookeeper:3.8.0'
   api "io.netty:netty-all:${versions.netty}"
+  api "org.eclipse.jetty:jetty-server:9.4.48.v20220622"
```
Comment (reta) [inline plugins/repository-hdfs/build.gradle:86]: I think we could exclude this one as well, is it really needed?

**Review by reta — APPROVED** [review]

**Pull request merged**